libs/cardano-ledger-core/src/Cardano/Ledger/Core.hs

```diff
@@ -425,6 +426,10 @@ toCompactPartial :: (HasCallStack, Val a) => a -> CompactForm a
 toCompactPartial v =
   fromMaybe (error $ "Illegal value in TxOut: " <> show v) $ toCompact v
 
+-- A version of mkBasicTxOut, which has only a Coin (no multiAssets) for every EraTxOut era.
+mkCoinTxOut :: forall era. EraTxOut era => Addr (EraCrypto era) -> Coin -> TxOut era
+mkCoinTxOut add coinx = mkBasicTxOut add (modifyCoin (const coinx) zero)
```

Comment:
Just noticed that one of my previous suggestions wasn't applied, so I'll add it again.
```suggestion
mkCoinTxOut addr = mkBasicTxOut addr . inject
```